pallets/gear/src/tests.rs

```diff
@@ -724,32 +751,43 @@ fn delayed_send_program_message_payment() {
 
         // Gas should be reserved until message is holding.
         assert_eq!(Balances::reserved_balance(USER_1), delay_holding_fee);
-        let free_balance = Balances::free_balance(&USER_1) + Balances::reserved_balance(USER_1);
+        let total_balance = Balances::free_balance(USER_1) + Balances::reserved_balance(USER_1);
 
         // Run blocks to release message.
         run_to_block(delay + 2, None);
 
         let delayed_id = MessageId::generate_outgoing(proxy_msg_id, 0);
 
-        // Check that delayed task was created
+        // Check that delayed task was created.
         assert!(TaskPoolOf::<Test>::contains(
             &(delay + 3),
             &ScheduledTask::SendDispatch(delayed_id)
         ));
 
-        // Block where message processed
+        // Block where message processed.
         run_to_next_block(None);
 
+        // Check that last event is MessagesDispatched.
+        let last_event = match System::events().last().map(|r| r.event.clone()) {
+            Some(MockRuntimeEvent::Gear(e)) => e,
+            _ => unreachable!("Should be one Gear event"),
+        };
+        match last_event {
+            Event::MessagesDispatched { .. } => {}
```

Comment:
assert amount within the block maybe? 
`assert_last_dequeued`

Reply:
done